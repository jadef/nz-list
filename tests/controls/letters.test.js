import React from 'react';
import renderer from 'react-test-renderer';

// Link to component(s)
import ControlsLetters from '../../app/js/components/controls/Letters';
import Letter from '../../app/js/components/controls/Letters';

// Describe
test('ControlsLetters Renders', () => {
  const component = renderer.create(
    <ControlsLetters />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Letter Renders', () => {
  const component = renderer.create(
    <Letter />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
