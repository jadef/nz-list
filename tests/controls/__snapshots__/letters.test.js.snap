// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsLetters Renders 1`] = `
<section
  className="letters controls"
>
  <div
    className="letter-wrapper"
  >
    <ul>
      <li
        className=" active"
      >
        <a
          href="#jumpA"
          onClick={[Function]}
        >
          A
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpB"
          onClick={[Function]}
        >
          B
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpC"
          onClick={[Function]}
        >
          C
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpD"
          onClick={[Function]}
        >
          D
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpE"
          onClick={[Function]}
        >
          E
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpF"
          onClick={[Function]}
        >
          F
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpG"
          onClick={[Function]}
        >
          G
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpH"
          onClick={[Function]}
        >
          H
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpI"
          onClick={[Function]}
        >
          I
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpJ"
          onClick={[Function]}
        >
          J
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpK"
          onClick={[Function]}
        >
          K
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpL"
          onClick={[Function]}
        >
          L
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpM"
          onClick={[Function]}
        >
          M
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpN"
          onClick={[Function]}
        >
          N
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpO"
          onClick={[Function]}
        >
          O
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpP"
          onClick={[Function]}
        >
          P
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpQ"
          onClick={[Function]}
        >
          Q
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpR"
          onClick={[Function]}
        >
          R
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpS"
          onClick={[Function]}
        >
          S
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpT"
          onClick={[Function]}
        >
          T
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpU"
          onClick={[Function]}
        >
          U
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpV"
          onClick={[Function]}
        >
          V
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpW"
          onClick={[Function]}
        >
          W
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpX"
          onClick={[Function]}
        >
          X
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpY"
          onClick={[Function]}
        >
          Y
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpZ"
          onClick={[Function]}
        >
          Z
        </a>
      </li>
    </ul>
    <div
      className="handle"
      id="letter-handle"
    >
      <svg
        viewBox="0 0 32 32"
      >
        <defs>
          <filter
            filterUnits="objectBoundingBox"
            height="210.0%"
            id="triangle-shadow"
            width="210.0%"
            x="-55.0%"
            y="-35.0%"
          >
            <feOffset
              dx="0"
              dy="4"
              in="SourceAlpha"
              result="shadowOffsetOuter1"
            />
            <feGaussianBlur
              in="shadowOffsetOuter1"
              result="shadowBlurOuter1"
              stdDeviation="3"
            />
            <feColorMatrix
              in="shadowBlurOuter1"
              type="matrix"
              values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.15 0"
            />
          </filter>
        </defs>
        <g
          transform="translate(10, 10) rotate(-90) translate(-10, -10) "
        >
          <use
            className="c-bg-dark"
            fillOpacity="1"
            filter="url(#triangle-shadow)"
            xlinkHref="#triangle"
          />
          <polygon
            className="c-bg-primary"
            id="triangle"
            points="10 0 20 20 0 20"
          />
        </g>
      </svg>
    </div>
  </div>
</section>
`;

exports[`Letter Renders 1`] = `
<section
  className="letters controls"
>
  <div
    className="letter-wrapper"
  >
    <ul>
      <li
        className=" active"
      >
        <a
          href="#jumpA"
          onClick={[Function]}
        >
          A
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpB"
          onClick={[Function]}
        >
          B
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpC"
          onClick={[Function]}
        >
          C
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpD"
          onClick={[Function]}
        >
          D
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpE"
          onClick={[Function]}
        >
          E
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpF"
          onClick={[Function]}
        >
          F
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpG"
          onClick={[Function]}
        >
          G
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpH"
          onClick={[Function]}
        >
          H
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpI"
          onClick={[Function]}
        >
          I
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpJ"
          onClick={[Function]}
        >
          J
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpK"
          onClick={[Function]}
        >
          K
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpL"
          onClick={[Function]}
        >
          L
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpM"
          onClick={[Function]}
        >
          M
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpN"
          onClick={[Function]}
        >
          N
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpO"
          onClick={[Function]}
        >
          O
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpP"
          onClick={[Function]}
        >
          P
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpQ"
          onClick={[Function]}
        >
          Q
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpR"
          onClick={[Function]}
        >
          R
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpS"
          onClick={[Function]}
        >
          S
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpT"
          onClick={[Function]}
        >
          T
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpU"
          onClick={[Function]}
        >
          U
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpV"
          onClick={[Function]}
        >
          V
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpW"
          onClick={[Function]}
        >
          W
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpX"
          onClick={[Function]}
        >
          X
        </a>
      </li>
      <li
        className=""
      >
        <a
          href="#jumpY"
          onClick={[Function]}
        >
          Y
        </a>
      </li>
      <li
        className="none"
      >
        <a
          href="#jumpZ"
          onClick={[Function]}
        >
          Z
        </a>
      </li>
    </ul>
    <div
      className="handle"
      id="letter-handle"
    >
      <svg
        viewBox="0 0 32 32"
      >
        <defs>
          <filter
            filterUnits="objectBoundingBox"
            height="210.0%"
            id="triangle-shadow"
            width="210.0%"
            x="-55.0%"
            y="-35.0%"
          >
            <feOffset
              dx="0"
              dy="4"
              in="SourceAlpha"
              result="shadowOffsetOuter1"
            />
            <feGaussianBlur
              in="shadowOffsetOuter1"
              result="shadowBlurOuter1"
              stdDeviation="3"
            />
            <feColorMatrix
              in="shadowBlurOuter1"
              type="matrix"
              values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.15 0"
            />
          </filter>
        </defs>
        <g
          transform="translate(10, 10) rotate(-90) translate(-10, -10) "
        >
          <use
            className="c-bg-dark"
            fillOpacity="1"
            filter="url(#triangle-shadow)"
            xlinkHref="#triangle"
          />
          <polygon
            className="c-bg-primary"
            id="triangle"
            points="10 0 20 20 0 20"
          />
        </g>
      </svg>
    </div>
  </div>
</section>
`;
